"featureUtility installServerFeatures" stopped working on 26.0.0.6 when FEATURE_REPO_URL is specified

**Describe the bug**  

We consume the container image `icr.io/appcafe/websphere-liberty:kernel-java21-openj9-ubi-minimal` to containerize our application. The Dockerfile runs a step like this:
```
  export FEATURE_REPO_USER=$(cat /run/secrets/TAAS_USER) && \
  export FEATURE_REPO_PASSWORD=$(cat /run/secrets/TAAS_PASSWORD) && \
  export FEATURE_REPO_URL=https://na.artifactory.swg-devops.com:443/artifactory/hyc-waca-maven-virtual && \
  features.sh
```

The step has been working fine till yesterday. It stopped working today ( Jun 16th, 2026 ) by an error like this:
```
#40 2.304 com.ibm.ws.install.InstallException: CWWKF1248E: One of the repositories that is specified in the configuration cannot be reached. Verify that your computer has network access and firewalls are configured correctly, then try the action again. If the connection still fails, the repository server might be temporarily unavailable.
```

I checked the repository access/availability and it seems fine. When I add `--verbose` option. I got this stacktrace:
```
#40 2.442 [6/17/26, 2:24:40:119 GMT]com.ibm.ws.install.InstallException: CWWKF1248E: One of the repositories that is specified in the configuration cannot be reached. Verify that your computer has network access and firewalls are configured correctly, then try the action again. If the connection still fails, the repository server might be temporarily unavailable.
#40 2.442 java.lang.RuntimeException: com.ibm.ws.install.InstallException: CWWKF1248E: One of the repositories that is specified in the configuration cannot be reached. Verify that your computer has network access and firewalls are configured correctly, then try the action again. If the connection still fails, the repository server might be temporarily unavailable.
#40 2.456 	at com.ibm.ws.install.internal.InstallKernelMap.checkWorkingRepository(InstallKernelMap.java:1504)
#40 2.459 	at com.ibm.ws.install.internal.InstallKernelMap.getMavenRepo(InstallKernelMap.java:1475)
#40 2.460 	at com.ibm.ws.install.internal.InstallKernelMap.downloadArtifact(InstallKernelMap.java:1365)
#40 2.460 	at com.ibm.ws.install.internal.InstallKernelMap.get(InstallKernelMap.java:310)
#40 2.462 	at com.ibm.ws.install.internal.InstallKernelMap.getJsonsFromMavenCentral(InstallKernelMap.java:1925)
#40 2.463 	at com.ibm.ws.install.featureUtility.FeatureUtility.getJsonFiles(FeatureUtility.java:789)
#40 2.466 	at com.ibm.ws.install.featureUtility.FeatureUtility.<init>(FeatureUtility.java:186)
#40 2.468 	at com.ibm.ws.install.featureUtility.FeatureUtility.<init>(FeatureUtility.java:52)
#40 2.469 	at com.ibm.ws.install.featureUtility.FeatureUtility$FeatureUtilityBuilder.build(FeatureUtility.java:927)
#40 2.470 	at com.ibm.ws.install.featureUtility.cli.InstallServerAction.install(InstallServerAction.java:269)
#40 2.472 	at com.ibm.ws.install.featureUtility.cli.InstallServerAction.execute(InstallServerAction.java:293)
#40 2.473 	at com.ibm.ws.install.featureUtility.cli.InstallServerAction.handleTask(InstallServerAction.java:84)
#40 2.475 	at com.ibm.ws.install.featureUtility.cli.FeatureAction.handleTask(FeatureAction.java:98)
#40 2.477 	at com.ibm.ws.install.featureUtility.FeatureUtilityExecutor.main(FeatureUtilityExecutor.java:60)
#40 2.479 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(Unknown Source)
#40 2.481 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
#40 2.482 	at com.ibm.ws.kernel.boot.cmdline.UtilityMain.internal_main(UtilityMain.java:175)
#40 2.484 	at com.ibm.ws.kernel.boot.cmdline.UtilityMain.main(UtilityMain.java:55)
#40 2.486 	at com.ibm.ws.kernel.boot.cmdline.Main.main(Main.java:54)
#40 2.487 Caused by: com.ibm.ws.install.InstallException: CWWKF1248E: One of the repositories that is specified in the configuration cannot be reached. Verify that your computer has network access and firewalls are configured correctly, then try the action again. If the connection still fails, the repository server might be temporarily unavailable.
#40 2.499 	at com.ibm.ws.install.internal.InstallKernelMap.checkWorkingRepository(InstallKernelMap.java:1501)
#40 2.503 	... 18 more
```

I found there's suspicious PR https://github.com/OpenLiberty/open-liberty/pull/34805 which seems to block any custom repository by the exception above.

I also tested one prior version ( 26.0.0.5 ) container image - `icr.io/appcafe/websphere-liberty:26.0.0.5-kernel-java21-openj9-ubi-minimal` and the features.sh script went successful on `26.0.0.5`.

**Steps to Reproduce**  
- run `featureUtility installServerFeatures` with FEATURE_REPO_URL environment variable on liberty 26.0.0.6

**Expected behavior**  
- features are downloaded successfully form the specified maven repository mirror.

**Diagnostic information:**  
 - OpenLiberty Version: 26.0.0.6
 - Affected feature(s)  featureUtility / features.sh
 - Java Version: java21
 - server.xml configuration (WITHOUT sensitive information like passwords)
 - If it would be useful, upload the messages.log file found in `$WLP_OUTPUT_DIR/messages.log`

**Additional context**  
Add any other context about the problem here.

--- a/dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java
+++ b/dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java
@@ -1495,10 +1495,12 @@ public class InstallKernelMap implements Map {
         try {
             Properties properties = RepositoryConfigUtils.loadRepoProperties();
 
-            if (workingRepos.isEmpty() && RepositoryConfigUtils.isWlpRepoEnabled(properties)) {
-                workingRepos.add(MAVEN_CENTRAL_REPOSITORY);
-            } else {
-                throw new InstallException(Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("ERROR_FAILED_TO_CONNECT_REPOS"));
+            if (workingRepos.isEmpty() ) {
+                if (RepositoryConfigUtils.isWlpRepoEnabled(properties)) {
+                    workingRepos.add(MAVEN_CENTRAL_REPOSITORY);
+                } else {
+                    throw new InstallException(Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("ERROR_FAILED_TO_CONNECT_REPOS"));
+                }
             }
         } catch (InstallException e) {
             throw new RuntimeException(e);

